gcloud-java-dns/src/main/java/com/google/gcloud/testing/LocalDnsHelper.java

```diff
@@ -0,0 +1,1426 @@
+        if ((name == null || name.equals(rrset.getName()))
+            && (type == null || type.equals(rrset.getType()))) {
+          return rrset;
+        }
+      }
+    }
+    return null;
+  }
+
+  /**
+   * We only provide the most basic validation for A and AAAA records.
+   */
+  static boolean checkRrData(String data, String type) {
+    // todo add validation for other records
+    String[] tokens;
+    switch (type) {
+      case "A":
+        tokens = data.split("\\.");
+        if (tokens.length != 4) {
+          return false;
+        }
+        for (String token : tokens) {
+          try {
+            Integer number = Integer.valueOf(token);
+            if (number < 0 || number > 255) {
+              return false;
+            }
+          } catch (NumberFormatException ex) {
+            return false;
+          }
+        }
+        return true;
+      case "AAAA":
+        tokens = data.split(":", -1);
+        if (tokens.length != 8) {
+          return false;
+        }
+        for (String token : tokens) {
+          try {
+            if (!token.isEmpty()) {
+              // empty is ok
+              Long number = Long.parseLong(token, 16);
+              if (number < 0 || number > 0xFFFF) {
+                return false;
+              }
+            }
+          } catch (NumberFormatException ex) {
+            return false;
+          }
+        }
+        return true;
+      default:
+        return true;
+    }
+  }
+
+  /**
+   * Check supplied listing options.
+   */
+  static Response checkListOptions(Map<String, Object> options) {
+    // for general listing
+    String maxResultsString = (String) options.get("maxResults");
+    if (maxResultsString != null) {
+      Integer maxResults = null;
+      try {
+        maxResults = Integer.valueOf(maxResultsString);
+      } catch (NumberFormatException ex) {
+        return Error.INVALID.response(String.format(
+            "Invalid integer value': '%s'.", maxResultsString));
+      }
+      if (maxResults <= 0) {
+        return Error.INVALID.response(String.format(
+            "Invalid value for 'parameters.maxResults': '%s'", maxResults));
+      }
+    }
+    String dnsName = (String) options.get("dnsName");
+    if (dnsName != null) {
+      if (!dnsName.endsWith(".")) {
+        return Error.INVALID.response(String.format(
+            "Invalid value for 'parameters.dnsName': '%s'", dnsName));
+      }
+    }
+    // for listing dns records, name must be fully qualified
+    String name = (String) options.get("name");
+    if (name != null) {
+      if (!name.endsWith(".")) {
+        return Error.INVALID.response(String.format(
+            "Invalid value for 'parameters.name': '%s'", name));
+      }
+    }
+    String type = (String) options.get("type"); // must be provided with name
+    if (type != null) {
+      if (name == null) {
+        return Error.INVALID.response("Invalid value for 'parameters.name': ''");
```

Comment:
I think it might be nice to include in the error message that name must be specified when type is specified.